autorest/azure/persist_test.go

```diff
@@ -0,0 +1,132 @@
+package azure
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+const MockTokenJSON string = `{
+	"access_token": "accessToken",
+	"refresh_token": "refreshToken",
+	"expires_in": "1000",
+	"expires_on": "2000",
+	"not_before": "3000",
+	"resource": "resource",
+	"token_type": "type"
+}`
+
+var TestToken = Token{
+	AccessToken:  "accessToken",
+	RefreshToken: "refreshToken",
+	ExpiresIn:    "1000",
+	ExpiresOn:    "2000",
+	NotBefore:    "3000",
+	Resource:     "resource",
+	Type:         "type",
+}
+
+func TestLoadToken(t *testing.T) {
+	f, err := ioutil.TempFile(os.TempDir(), "testloadtoken")
+	if err != nil {
+		t.Errorf("unexpected error when creating temp file: %v", err)
+	}
+	defer os.Remove(f.Name())
+
+	_, err = f.Write([]byte(MockTokenJSON))
+	if err != nil {
+		t.Errorf("unexpected error when writing temp test file: %v", err)
+	}
+
+	expectedToken := TestToken
+	actualToken, err := LoadToken(f.Name())
+	if err != nil {
+		t.Errorf("unexpected error loading token from file: %v", err)
+	}
+
+	if *actualToken != expectedToken {
+		t.Errorf("failed to decode properly actual(%v) expected(%v)", *actualToken, expectedToken)
+	}
+}
```

Comment:
@brendandixon that's idiomatic, I don't see a problem with that. t.Fatal/t.Error difference is exactly for this purpose.